pytorch_toolkit/formula_recognition/im2latex/models/backbones/resnet.py

```diff
@@ -29,13 +29,9 @@
 
 
 class ResNetLikeBackbone(nn.Module):
-    def __init__(self, configuration):
+    def __init__(self, disable_layer_3, disable_layer_4, arch, in_lstm_ch = 512, enable_last_conv = False):
         super(ResNetLikeBackbone, self).__init__()
-        disable_layer_3 = configuration.get('disable_layer_3')
-        disable_layer_4 = configuration.get('disable_layer_4')
-        arch = configuration.get('arch')
-        self.in_lstm_ch = configuration.get('in_lstm_ch', 512)
-        enable_last_conv = configuration.get('enable_last_conv', False)
+        self.in_lstm_ch = in_lstm_ch
```

Comment:
`in_lstm_ch` is probably not the best name -- it refers to the context that is output of the scope of this particular component. In other words, at the whole model scale it makes perfect sense, but backbone on its own should not rely on the latter stages of the model. My suggestion here is to rename it to something neutral, like `output_channels`, and maybe check that it is in compliance with the actual output of the backbone, in case of the last conv absence. Does it make sense?

Reply:
Renamed parameter & added check in the model class that number of the output channels of the backbone is equal to number of the input channels in the head